patch-package
--- a/node_modules/promise-request-retry/index.js
+++ b/node_modules/promise-request-retry/index.js
@@ -1,7 +1,7 @@
 'use strict';
 const requestPromise = require('request-promise');
 const Promise = require('bluebird');
-const logger = require('./modules/logger')('request-promise-retry');
+const logger  = require('winston');
 
 class rpRetry {
     static _rpRetry(options) {
@@ -22,7 +22,16 @@ class rpRetry {
                     logger.info(`Encountered error ${err.message} for ${options.method} request to ${options.uri}, retry count ${tryCount}`);
                     tryCount -= 1;
                     if (tryCount) {
-                        return fetchDataWithRetry(tryCount);
+                        let delay = options.delay || 100; // default ms delay between retries
+                        if (options.factor) {
+                            delay *= options.factor;
+                        }
+                        return new Promise((resolve, reject) => {
+                            setTimeout(() => {
+                                logger.debug(`waiting for ${delay} ms before next retry for ${options.uri}`);
+                                resolve(fetchDataWithRetry(tryCount));
+                            }, delay);
+                        });
                     }
                     return Promise.reject(err);
                 });
